mico-admin/src/app/api/api.service.ts

```diff
@@ -249,24 +257,47 @@ export class ApiService {
             return;
         }
 
-
         const resource = 'services/';
 
-        return this.rest.post(resource, data).pipe(flatMap(val => {
+        return this.rest.post<ApiObject>(resource, data).pipe(flatMap(val => {
 
-            const stream = this.getStreamSource(val._links.self.href);
+            const stream = this.getStreamSource<ApiObject>(val._links.self.href);
             stream.next(val);
 
             this.getServices();
             this.getServiceVersions(data.shortName);
 
-            return (stream.asObservable() as Observable<Readonly<ApiObject>>).pipe(
+            return stream.asObservable().pipe(
                 filter(service => service !== undefined)
             );
         }));
 
     }
 
+    postServiceViaGithub(data): Observable<Readonly<ApiObject>> {
+        if (data == null) {
+            return;
+        }
+
+        const resource = 'services/import/github';
```

Comment:
remove variable if unused for better clarity. This variable doesn't hold the resource string used to get the stream as this value can only be calculated after the post finished.